jupyterlab_nvdashboard/apps/amd_gpu.py

```diff
@@ -0,0 +1,83 @@
+# requires rocm!
+import subprocess
+import logging
+import re
+
+
+logging.basicConfig(
+    filename="/var/log/amd_gpu.log",
+    level=logging.INFO,
+    format="%(asctime)s %(levelname)s %(name)s %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+
+class AmdGpuProperties:
+    def __init__(self, bash=subprocess):
+        self.bash = bash
+        self.gpus = self.get_gpu_count()
+
+    def get_gpu_count(self) -> None:
+        """Get the number of working GPUs."""
+        try:
+            output = self.bash.run("rocm-smi", capture_output=True).stdout
+            if "ROCm System Management Interface" in output:
+                gpus_count = [int(num[-1]) for num in re.findall(r'\n\d{1,2}', str(output))]
+                self.gpus = len(gpus_count)
+            else:
+                self.gpus = 0
+
+        except Exception:
+            logger.error("An error has occured while searching for the number of GPUs!")
+
+    def get_gpu_clock_freq(self, flag="-g") -> list[int]:
+        """Return the clock frequence of each GPU in Mhz."""
+        try:
+            output = self.bash.run(["rocm-smi", flag], capture_output=True).stdout
+            if "ROCm System Management Interface" in output:
+                freq = [int(num[1:-1]) for num in re.findall(r'\([0-9]*M', str(output))]
+                return freq
+            else:
+                return [0 for _ in range(self.gpus)]
+
+        except Exception:
+            logger.error("An error has occured while searching for the clock frequence!")
+
+    def get_gpu_mem_use(self, flag="--showmemuse") -> list[int]:
+        """Return the current memory usage of each GPU in %."""
+        try:
+            output = self.bash.run(["rocm-smi", flag], capture_output=True).stdout
+            if "ROCm System Management Interface" in output:
+                memUse = [int(num[0:-1]) for num in re.findall(r'\d{1,3}\n', str(output))]
+                return memUse
+            else:
+                return [0 for _ in range(self.gpus)]
+
+        except Exception:
+            logger.error("An error has occured while searching for the memory usage!")
+    
+    def get_gpu_pcie_bandwith(self, flag="-b") -> list[float]:
```

Comment:
We want to use the type not the value so its capital you may want to import it from "from typing import List"

``
```suggestion
    def get_gpu_pcie_bandwith(self, flag="-b") -> List[float]:
```